src/Sylius/Bundle/CoreBundle/Installer/Provider/DatabaseSetupCommandsProvider.php

```diff
@@ -26,13 +28,27 @@
 
 final class DatabaseSetupCommandsProvider implements DatabaseSetupCommandsProviderInterface
 {
-    private bool $isPostgreSQLPlatform;
+    /** @var AbstractSchemaManager<PostgreSQLPlatform|MySQLPlatform>|null */
+    private ?AbstractSchemaManager $schemaManager = null;
+
+    public function __construct(
+        private Registry $doctrineRegistry,
+        private ?EntityManagerInterface $entityManager = null,
```

Comment:
```suggestion
        private Registry|EntityManagerInterface $entityManager,
```
and a deprecation notice done on `instanceof` could ease potential migration

Reply:
@Rafikooo, please consider this (and fix it if you decide to) in a separate pull request.

Reply:
https://github.com/Sylius/Sylius/pull/15427